src/plugins/discover/public/application/components/doc_viewer/doc_viewer_render_error.tsx

```diff
@@ -7,20 +7,18 @@
  */
 
 import React from 'react';
-import { EuiCallOut, EuiCodeBlock } from '@elastic/eui';
-import { formatMsg, formatStack } from '../../../../../kibana_legacy/public';
+import { EuiErrorBoundary } from '@elastic/eui';
 
 interface Props {
   error: Error | string;
 }
 
-export function DocViewerError({ error }: Props) {
-  const errMsg = formatMsg(error);
-  const errStack = typeof error === 'object' ? formatStack(error) : '';
+const DocViewerErrorWrapper = ({ error }: Props) => {
+  throw error;
+};
 
-  return (
-    <EuiCallOut title={errMsg} color="danger" iconType="cross" data-test-subj="docViewerError">
-      {errStack && <EuiCodeBlock>{errStack}</EuiCodeBlock>}
```

Comment:
I like idea of using `EuiErrorBoundary` instead of `EuiCodeBlock` / `CodeEditor`. Thank you